### PR TITLE
Include thread replies in message previews

### PR DESCRIPTION
**Pull request by weeman1337**
:information_source: should be reviewed with ignored whitespaces.

image

closes https://github.com/vector-im/element-web/issues/23920

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

element-web notes: Thread replies are now also displayed in message previews

PSF-1991

---
Here's what your changelog entry will look like:

## ✨ Features
 * Include thread replies in message previews ([\#10631](https://github.com/matrix-org/matrix-react-sdk/pull/10631)). Fixes vector-im/element-web#23920.

### CODE REVIEW

File: test/components/views/rooms/RoomTile-test.tsx [hunk 1]
```diff
@@ -99,130 +108,199 @@ describe("RoomTile", () => {
...
+        describe("and there is a message in a thread", () => {
+            beforeEach(() => {
+                const message = mkMessage({
+                    event: true,
+                    room: room.roomId,
+                    msg: "test thread reply",
+                    user: client.getSafeUserId(),
+                });
+
+                // Mock thread reply for tests.
```
Comment (weeman1337) [inline test/components/views/rooms/RoomTile-test.tsx:291]: If anyone knows how to properly add a thread in a more „integration test“ fashioned way, please leave a comment.

File: test/components/views/rooms/RoomTile-test.tsx [hunk 2]
```diff
@@ -79,12 +87,44 @@ describe("RoomTile", () => {
...
+            event: true,
+            room: room.roomId,
+            msg: "test thread reply",
+            user: client.getSafeUserId(),
+            ts,
+        });
+
+        // Mock thread reply for tests.
+        jest.spyOn(room, "getThreads").mockReturnValue([
+            // @ts-ignore
```
Comment (weeman1337) [inline test/components/views/rooms/RoomTile-test.tsx:120]: If anyone knows how to properly add a thread in a more „integration test“ fashioned way, please leave a comment.

**Comment by weeman1337** [comment]
Reactions are still an issue. Waiting for https://github.com/matrix-org/matrix-react-sdk/pull/10601

File: res/css/views/rooms/_RoomTile.pcss [hunk 3]
```diff
@@ -55,13 +55,17 @@ limitations under the License.
...
+            align-items: center;
+            color: $secondary-content;
+            display: flex;
+            gap: $spacing-4;
+            line-height: $font-18px;
+        }
 
-            /* Ellipsize any text overflow */
-            text-overflow: ellipsis;
+        .mx_RoomTile_subtitle_text {
```
Comment (justjanne) [inline res/css/views/rooms/_RoomTile.pcss:66]: Long titles don't properly overflow with an ellipsis anymore, instead they break into the next line. The title also needs to be prevented from wrapping or overflowing, as it was before.

```suggestion
        .mx_RoomTile_title,
        .mx_RoomTile_subtitle_text {
```

File: res/img/compound/thread-16px.svg [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path fill-rule="evenodd" clip-rule="evenodd" d="M2.88867 0.138672C1.36989 0.138672 0.138672 1.36989 0.138672 2.88867V15.1109C0.138672 15.395 0.299174 15.6547 0.553262 15.7817C0.80735 15.9088 1.11141 15.8813 1.33867 15.7109L4.36089 13.4442C4.57726 13.2819 4.84043 13.1942 5.11089 13.1942H13.1109C14.6297 13.1942 15.8609 11.963 15.8609 10.4442V2.88822C15.8609 1.36922 14.6295 0.138672 13.1109 0.138672H2.88867ZM3.69421 5.33301C3.69421 4.91879 4.03 4.58301 4.44421 4.58301H11.5553C11.9695 4.58301 12.3053 4.91879 12.3053 5.33301C12.3053 5.74722 11.9695 6.08301 11.5553 6.08301H4.44421C4.03 6.08301 3.69421 5.74722 3.69421 5.33301ZM4.44421 7.24976C4.03 7.24976 3.69421 7.58554 3.69421 7.99976C3.69421 8.41397 4.03 8.74976 4.44421 8.74976H7.99977C8.41398 8.74976 8.74977 8.41397 8.74977 7.99976C8.74977 7.58554 8.41398 7.24976 7.99977 7.24976H4.44421Z" fill="#737D8C"/>
```
Comment (justjanne) [inline res/img/compound/thread-16px.svg:2]: In dark mode, this icon should have have the color `#a9b2bc`, just like e.g., the explore icon next to the spotlight search. `fill="currentColor"` is the elegant way to solve this.

```suggestion
<path fill-rule="evenodd" clip-rule="evenodd" d="M2.88867 0.138672C1.36989 0.138672 0.138672 1.36989 0.138672 2.88867V15.1109C0.138672 15.395 0.299174 15.6547 0.553262 15.7817C0.80735 15.9088 1.11141 15.8813 1.33867 15.7109L4.36089 13.4442C4.57726 13.2819 4.84043 13.1942 5.11089 13.1942H13.1109C14.6297 13.1942 15.8609 11.963 15.8609 10.4442V2.88822C15.8609 1.36922 14.6295 0.138672 13.1109 0.138672H2.88867ZM3.69421 5.33301C3.69421 4.91879 4.03 4.58301 4.44421 4.58301H11.5553C11.9695 4.58301 12.3053 4.91879 12.3053 5.33301C12.3053 5.74722 11.9695 6.08301 11.5553 6.08301H4.44421C4.03 6.08301 3.69421 5.74722 3.69421 5.33301ZM4.44421 7.24976C4.03 7.24976 3.69421 7.58554 3.69421 7.99976C3.69421 8.41397 4.03 8.74976 4.44421 8.74976H7.99977C8.41398 8.74976 8.74977 8.41397 8.74977 7.99976C8.74977 7.58554 8.41398 7.24976 7.99977 7.24976H4.44421Z" fill="currentColor"/>
```

**Review by justjanne — CHANGES_REQUESTED** [review]

**Review by justjanne — APPROVED** [review]

**Pull request merged**